src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -119,20 +120,39 @@ public override void Abort()
         /// <summary>Starts this system</summary>
         public void Start()
         {
-            _provider.Init(this);
-            ConfigureLoggers();
-            LoadExtensions();
+            try
+            {
+                RegisterOnTermination(StopScheduler);
```

Comment:
Sets a `TerminationHook` for ensuring that the `IScheduler` is terminated before `ActorSystem.WhenTerminated` completes, which guarantees that we won't leak any of the `IScheduler` threads across multiple specs in the test suite.